libff/CMakeLists.txt

```diff
@@ -24,13 +24,20 @@ file(
   GLOB_RECURSE
   LIBFF_SOURCE
 
-  algebra/curves/alt_bn128/*.cpp
-  algebra/curves/edwards/*.cpp
-  algebra/curves/mnt/mnt4/*.cpp
-  algebra/curves/mnt/mnt46_common.cpp
-  algebra/curves/mnt/mnt6/*.cpp
-  algebra/curves/bls12_377/*.cpp
-  algebra/curves/bw6_761/*.cpp
+  algebra/curves/alt_bn128/*.?pp
+  algebra/curves/alt_bn128/*.tcc
+  algebra/curves/edwards/*.?pp
+  algebra/curves/edwards/*.tcc
+  algebra/curves/mnt/mnt4/*.?pp
+  algebra/curves/mnt/mnt4/*.tcc
+  algebra/curves/mnt/mnt46_common.?pp
+  algebra/curves/mnt/mnt46_common.tcc
+  algebra/curves/mnt/mnt6/*.?pp
```

Comment:
Why did you add the hpp and tcc to the LIBFF_SOURCE variable?

Reply:
I was having some issues where things were not rebuilding correctly in incremental builds.  I think this adds more dependencies to the makefile (i.e. so it knows to rebuild, say, a test if some .tcc changes).

Reply:
Got it.
Shall we do:

```
algebra/curves/**.?pp algebra/curves/**.tcc
```
etc. instead of listing the curves here (all supported curves are added here, so the shortcut notation above is equivalent)

Reply:
Ah, yes.  I didn't mention I originally tried this, but it brought in all the profile and test files.  It started to get awkward to correctly filter them all.  In a separate change we should probably consistently keep those non lib files in their own dirs.

Reply:
Ah yes of course... I see. Alright let's stick to that then!